src/smirnoff_hack.py

```diff
@@ -62,4 +62,4 @@ def getForceField(*ffpaths):
     cache_key = hasher.hexdigest()
     if cache_key not in SMIRNOFF_FORCE_FIELD_CACHE:
         SMIRNOFF_FORCE_FIELD_CACHE[cache_key] = ForceField(*ffpaths, allow_cosmetic_attributes=True)
```

Comment:
This is really clever. Perhaps related -- if it becomes necessary to _compare_ force fields in the future, I think we could do it using something like a hash of the output of `ff.to_string('XML')`. That should standardize whitespace and _I think_ that recent changes may ensure that parameters are written out in a consistent order. Floating point values may still be trouble, but, if this sort of thing ever became important, I could put some time into ensuring that comparison works.

Reply:
I think a good discussion on how to compare force fields is well worth it. I'd like to see us eventually have the ability to apply a parameter change to a system, and compute energies at a fraction of the cost of recomputing the energy of the whole system.

Reply:
Agreed -- I think @simonboothroyd is also looking for this functionality. Hopefully @andrearizzi's work on a dedicated OpenFF `System` class could get us closer to this as well.